docs/latest/design/request-authentication.md

```diff
@@ -0,0 +1,537 @@
+# Request Authentication
+
+## Overview
+
+[Issue 336][] specifies the need for exposing a user-facing API to configure request authentication. Request
+authentication is defined as an authentication mechanism to be enforced by Envoy on a per-request basis. A connection
+will be rejected if it contains invalid authentication information, based on the `Authentication` API type proposed in
+this design document.
+
+Envoy Gateway leverages [Gateway API][] for configuring managed Envoy proxies. Gateway API defines core, extended, and
+implementation-specific API [support levels][] for implementors such as Envoy Gateway to expose features. Since
+implementing request authentication is not covered by `Core` or `Extended` APIs, an `Implementation-specific` API will
+be created for this purpose.
+
+## Goals
+
+* Define an API for configuring request authentication.
+* Implement [JWT] as the first supported authentication type.
+* Allow users that manage routes, e.g. [HTTPRoute][], to authenticate matching requests before forwarding to a backend
+  service.
+* Support HTTPRoutes as an Authentication API referent. HTTPRoute provides multiple [extension points][]. The
+  [HTTPRouteFilter][] is the extension point supported by the Authentication API.
+
+## Non-Goals
+
+* Allow infrastructure administrators to override or establish default authentication policies.
+* Support referents other than HTTPRoute.
+* Support Gateway API extension points other than HTTPRouteFilter.
+
+## Use-Cases
+
+These use-cases are presented as an aid for how users may attempt to utilize the outputs of the design. They are not an
+exhaustive list of features for authentication support in Envoy Gateway.
+
+As a Service Producer, I need the ability to:
+* Authenticate a request before forwarding it to a backend service.
+* Have different authentication mechanisms per route rule.
+* Choose from different authentication mechanisms supported by Envoy, e.g. OIDC.
+
+### Security API Group
+
+A new API group, `security.gateway.envoyproxy.io` is introduced to group security-related APIs. This will allow security
```

Comment:
curious why a new group is needed ?
why not graduate individual extensions from `v1alpha1` to `v1beta1` similar to what Gateway API does https://github.com/kubernetes-sigs/gateway-api/tree/main/apis

Reply:
A group is not needed per se. The reason for the group is documented. I don't have a strong opinion on the topic. We grouped the config APIs, so we've already established a pattern.